src/validator/Tests.java

```diff
@@ -0,0 +1,55 @@
+package validator;
+
+import static org.junit.Assert.*;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
+import org.junit.Test;
+
+public class Tests {
```

Comment:
Test is not covering all paths when code is restructure to have separate class for validation it can have public methods which can be easily testes like GenneralCardValidaror with public methods checkLengthCorrectness, checkIINMaskCorrectness , checkLuhnalgorithmCorrectness, this class would be used for within CreditCardValidator as a member